From 363185b72ccb5279fa950a378ebd93c082078d39 Mon Sep 17 00:00:00 2001
From: Aaro Koskinen <aaro.koskinen@iki.fi>
Date: Sat, 15 Mar 2014 01:32:56 +0200
Subject: libkeymap: fix kmap on big-endian machines

Currently "loadkeys" does not work properly (no keys are ever changed)
on big-endian machines (e.g. SPARC and other classic hardware)
due to incorrect typing. The patch fixes this.

Signed-off-by: Aaro Koskinen <aaro.koskinen@iki.fi>

diff --git a/src/libkeymap/kmap.c b/src/libkeymap/kmap.c
index e529136..85a3bb3 100644
--- a/src/libkeymap/kmap.c
+++ b/src/libkeymap/kmap.c
@@ -37,7 +37,7 @@ int
 lk_key_exists(struct lk_ctx *ctx, unsigned int k_table, unsigned int k_index)
 {
 	struct lk_array *map;
-	u_short *key;
+	unsigned int *key;
 
 	map = lk_array_get_ptr(ctx->keymap, k_table);
 	if (!map) {
-- 
cgit v0.10.1

